smarctlide-monitoring/src/main/java/eu/smartclide/contexthandling/ServiceMain.java

```diff
@@ -0,0 +1,132 @@
+            smartclideConfigPath = smartclideHome + File.separator + "config";
+        // Linux config directory /var/lib/smartclide
+        } else if (Files.exists(Paths.get("/var/lib/smartclide"))) {
+            smartclideHome = "/var/lib/smartclide";
+            smartclideConfigPath = smartclideHome + File.separator + "config";
+        // Linux config directory /opt/smartclide/config
+        } else if (Files.exists(Paths.get("/opt/smartclide"))) {
+            smartclideHome = "/opt/smartclide";
+            smartclideConfigPath = smartclideHome + File.separator + "config";
+        // Windows COnfigu Directories
+        } else if (Files.exists(Paths.get("C:\\ProgramData\\smartclide"))) {
+            smartclideHome = "C:\\ProgramData\\smartclide";
+            smartclideConfigPath = smartclideHome + File.separator + "config";
+        }
+
+        AmIMonitoringConfiguration amionfig = new AmIMonitoringConfiguration();
+        amionfig.setId("SMARTCLIDE");
+        amionfig.setServiceConfiguration(readFile(smartclideConfigPath + File.separator + "monitoring-config.xml"));
+
+        File configFile = new File(
+                smartclideConfigPath + File.separator + "services-config.xml");
+        String filepath = configFile.getPath();
+        SWServiceContainer serviceContainer = new SWServiceContainer(
+                "AmI-repository", filepath);
+        ServiceManager.getLSWServiceContainer().add(serviceContainer);
+        server = ServiceManager.registerWebservice(serviceContainer);
+
+
+        for (SWServiceContainer container : ServiceManager.getLSWServiceContainer()) {
+            if (Objects.requireNonNull(container.getServerClass()).toString()
+                    .contains("AmIMonitoringDataRepository")) {
+                reposService = ServiceManager.getWebservice(container);
+            }
+        }
+
+        server = ServiceManager.registerWebservice(AmIMonitoringService.class);
+        service = ServiceManager.getWebservice(IAmIMonitoringService.class);
+        service.configureService(amionfig);
+    }
+
+    private static String readFile(String filename) {
+        File f = new File(filename);
+        try {
+            byte[] bytes = Files.readAllBytes(f.toPath());
+            return new String(bytes, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return "";
+    }
+
+    private static void startService() {
+        // start monitoring service (the repository is implicitly started from within the monitoring service)
+        try {
+            service.start();
+        } catch (ContextFault e) {
+            logger.error(e.getMessage(), e);
+        }
+
+        // wait a while (10s)
+        try {
+            Thread.sleep(10000L);
+        } catch (InterruptedException e) {
+            logger.error(e.getMessage(), e);
+        }
+    }
+
+    public static void main(String[] args) {
```

Comment:
Running this causes an Error:
`ERROR [File watch service Thread] (FilePairSystemMonitor.java:232) - path .../smartclide/smartclide-context/target/classes/filepairmonitor is not a directory`

Reply:
please check the change

Reply:
Now I get: `java.nio.file.NoSuchFileException: resources/monitoring-config.xml` and
```
ERROR [main] (SWService_Configuration_FileMngr.java:145) - Could not serialize the Service Configuration file: resources/services-config.xml
java.lang.IllegalArgumentException: Null InputStream is not a valid argument
```

Reply:
updated to check for existing config directory